CMakeLists.txt

```diff
@@ -133,6 +133,31 @@ endif()
 find_package(Threads REQUIRED)
 find_package(ZLIB REQUIRED)
 
+# try to find an installed boost or install locally with conan
+set(boost_VERSION "1.71")
+# find_package(Boost ${boost_VERSION} QUIET)
+# if (NOT Boost_FOUND)
+  # bail if there's no conan installed
+  message(STATUS "No compatible boost version detected, using conan...")
+  find_program(conan_FOUND conan)
+  if (NOT conan_FOUND)
+    message(FATAL_ERROR "conan needs to be installed for boost, see https://docs.conan.io/en/latest/installation.html.")
+  endif()
+
+  # build dir needs to be added, that's where conan writes some info
+  list(APPEND CMAKE_MODULE_PATH ${CMAKE_BINARY_DIR})
+  # list(APPEND CMAKE_PREFIX_PATH ${CMAKE_BINARY_DIR})
```

Comment:
I'm really not sure why this line should be needed. there's nothing being installed to the build directory that needs to be found with `find_package/program/xx` unless some magic in conan cmake is requiring it.. that's how the documentation uses conan's `cmake_find_package` generator. linux configs and builds fine without, lets see what CI says for osx & win

Reply:
I don't know either. Let's dance like the Conan folks do.

Reply:
yeah this line on azure cmake config might be related:
```
ERROR: Not able to automatically detect 'C:/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/VC/Tools/MSVC/14.29.30133/bin/Hostx64/x64/cl.exe' version
ERROR: Unable to find a working compiler
``` 
it's not actually erroring out though, it still installs fine (not sure why it'd need a compiler to install in the first place..). anyways, I'll go with the documented way and uncomment that line again